src/error-handling/try-operator.md

```diff
@@ -4,42 +4,47 @@ The try-operator `?` is used to return errors to the caller. It lets you turn
 the common
 
 ```rust,ignore
-match some_expression {
+let some_variable = match some_expression {
     Ok(value) => value,
     Err(err) => return Err(err),
-}
+};
 ```
 
 into the much simpler
 
 ```rust,ignore
-some_expression?
+let some_variable = some_expression?;
 ```
 
 We can use this to simplify our error handing code:
 
 ```rust,editable
 use std::fs;
-use std::io::{self, Read};
+use std::io::{self, Error, ErrorKind, Read};
 
-fn read_username(path: &str) -> Result<String, io::Error> {
-    let username_file_result = fs::File::open(path);
-    let mut username_file = match username_file_result {
+// Reads the email address from a file and returns the username and hostname.
+fn read_email_address(path: &str) -> Result<(String, String), io::Error> {
+    let email_file_result = fs::File::open(path);
+    let mut email_file = match email_file_result {
         Ok(file) => file,
         Err(err) => return Err(err),
     };
 
-    let mut username = String::new();
-    match username_file.read_to_string(&mut username) {
-        Ok(_) => Ok(username),
-        Err(err) => Err(err),
+    let mut email = String::new();
+    if let Err(err) = email_file.read_to_string(&mut email) {
+        return Err(err);
+    }
+    if let Some((username, hostname)) = email.split_once('@') {
+        Ok((String::from(username), String::from(hostname)))
+    } else {
+        Err(Error::new(ErrorKind::InvalidData, "Invalid email address"))
```

Comment:
I think this adds too much extra logic to the example. The point here is to quickly (in 30 seconds or so) transform the code into using `?`. I would like to just keep the speaker notes here.

Reply:
Shortening this now

Reply:
What do you think about the change?